src/actionCreators/search.js

```diff
@@ -8,33 +8,47 @@ import {
 } from "utilities/authorityConfig"
 import { addSearchHistory as addApiSearchHistory } from "actionCreators/user"
 import { addSearchHistory } from "actions/history"
+import { clearErrors, addError } from "actions/errors"
 
-export const fetchSinopiaSearchResults = (query, options) => (dispatch) =>
-  getSearchResultsWithFacets(query, options).then(
-    ([response, facetResponse]) => {
-      dispatch(addSearchHistory(sinopiaSearchUri, "Sinopia resources", query))
-      dispatch(addApiSearchHistory(sinopiaSearchUri, query))
-      dispatch(
-        setSearchResults(
-          "resource",
-          sinopiaSearchUri,
-          response.results,
-          response.totalHits,
-          facetResponse || {},
-          query,
-          options,
-          response.error
+export const fetchSinopiaSearchResults =
+  (query, options, errorKey) => (dispatch) => {
+    dispatch(clearErrors(errorKey))
+    return getSearchResultsWithFacets(query, options).then(
+      ([response, facetResponse]) => {
+        dispatch(addSearchHistory(sinopiaSearchUri, "Sinopia resources", query))
+        dispatch(addApiSearchHistory(sinopiaSearchUri, query))
+        dispatch(
+          setSearchResults(
+            "resource",
+            sinopiaSearchUri,
+            response.results,
+            response.totalHits,
+            facetResponse || {},
+            query,
+            options,
+            response.error
+          )
         )
-      )
-    }
-  )
+        if (response.error) {
+          dispatch(
+            addError(errorKey, [
```

Comment:
Switched from using errors that are stored as part of the search results to errors that are stored with the other others, hence dispatching `addError()`.